ui/dashboards/src/components/PanelDrawer/PanelEditorForm.tsx

```diff
@@ -24,37 +24,41 @@ import {
   TextField,
   Typography,
 } from '@mui/material';
-import { QueryDefinition, UnknownSpec } from '@perses-dev/core';
+import { PanelDefinition } from '@perses-dev/core';
 import { ErrorAlert, ErrorBoundary } from '@perses-dev/components';
-import { OptionsEditorProps, PluginKindSelect, PluginSpecEditor, usePluginEditor } from '@perses-dev/plugin-system';
+import { PluginKindSelect, usePluginEditor } from '@perses-dev/plugin-system';
 import { useListPanelGroups } from '../../context';
 import { PanelEditorValues } from '../../context/DashboardProvider/panel-editor-slice';
 import { PanelPreview } from './PanelPreview';
+import { PanelSpecEditor } from './PanelSpecEditor';
+import { usePanelEditor } from './usePanelEditor';
 
 export interface PanelEditorFormProps {
   initialValues: PanelEditorValues;
   onChange: (values: PanelEditorValues) => void;
 }
 
 export function PanelEditorForm(props: PanelEditorFormProps) {
-  const { initialValues, onChange } = props;
+  const {
+    initialValues: { panelDefinition: initialPanelDef, groupId: initialGroupId },
+    onChange,
+  } = props;
 
   const panelGroups = useListPanelGroups();
-
-  const [name, setName] = useState(initialValues.name);
-  const [description, setDescription] = useState(initialValues.description);
-  const [groupId, setGroupId] = useState(initialValues.groupId);
-  const [kind, setKind] = useState(initialValues.kind);
-  const [spec, setSpec] = useState(initialValues.spec);
-  const [queries, setQueries] = useState(initialValues.queries);
+  const [groupId, setGroupId] = useState(initialGroupId);
+  const { panelDefinition, setName, setDescription, setQueries, setPlugin, setPanelDefinition } =
```

Comment:
added `usePanelEditor` to help keep track of all the states we need since most child components need to know the full panel definition now